src/WorkflowCore/Services/WorkflowController.cs

```diff
@@ -83,6 +85,8 @@ public async Task<string> StartWorkflow<TData>(string workflowId, int? version,
 
             wf.ExecutionPointers.Add(_pointerFactory.BuildGenesisPointer(def));
 
+            await _middlewareRunner.RunPreMiddleware(wf);
```

Comment:
The ID of the `WorkflowInstance`? Yes. And with the way that the interface looks, they should have access to every property of the `WorkflowInstance` since it gets passed down to the `Handle` method.

Reply:
Yes, but the ID is generated when you first persist the workflow, which hasn't happened at this point?

Reply:
Ah, it is? Is the ID always a Guid? If so, we can rely on that. I was hoping to have it run before the persistence step so that any changes to the workflow (i.e. setting the description) would be persisted along with it.

Would it make sense to persist it once before and once after?

Reply:
It's not strictly a Guid... it's up to the implementation of the persistence provider...
If we are going to persist it and then do more work... we'd also probably need to hold a lock on the workflow ID, so that none of the workers pick it up and try to process it before we've finished.  There is potential for a race condition here.

Reply:
Hmm... that would add a bit of complexity to it. So how about we just document that the workflow does not have an id in the pre-workflow middleware and keep it prior to the initial persistence?

Reply:
Yeah, we can revisit this in future versions